src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,194 @@
-
-    starpositions = SkyCoord(
-        ra=Angle(catalog["RAJ2000"], unit=u.deg),
-        dec=Angle(catalog["DEJ2000"], unit=u.deg),
-        frame="icrs",
-        copy=False,
-    )
-    catalog["ra_dec"] = starpositions
-
-    if magnitude_cut is not None:
-        catalog = catalog[catalog["Vmag"] < magnitude_cut]
+    stars_ = deepcopy(stars)
+    if magnitude_cut:
+        try:
+            stars_ = stars_[stars_[band] < magnitude_cut]
+        except KeyError:
+            raise ValueError(
+                f"The requested catalogue has no compatible magnitude for the {band} band"
+            )
 
     if radius is not None:
-        if pointing is None:
+        if pointing:
+            stars_["separation"] = stars_["ra_dec"].separation(pointing)
+            stars_ = stars_[stars_["separation"] < radius]
+        else:
             raise ValueError(
                 "Sky pointing, pointing=SkyCoord(), must be "
                 "provided if radius is given."
             )
-        separations = catalog["ra_dec"].separation(pointing)
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    return stars_
+
+
+def get_star_catalog(catalog, magnitude_cutoff=8.0, row_limit=1000000):
+    """
+    Utility function to download a star catalog for the get_bright_stars function
+
+    Parameters
+    ----------
+    catalog: string
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalogues. Default: Yale
+    min_magnitude: float
+        Minimum value for magnitude used in lookup
+    max_magnitude: float
+        Maximum value for magnitude used in lookup
+    row_limit: int
+        Maximum number of rows for the star catalog lookup
+
+    Returns
+    ----------
+    Astropy table:
+       List of all stars after cuts with catalog numbers, magnitudes,
+       and coordinates as SkyCoord objects including proper motion
+    """
+    from astroquery.vizier import Vizier
+
+    catalog_dict = StarCatalogues[catalog].value
+
+    vizier = Vizier(
+        catalog=catalog_dict["directory"],
+        columns=catalog_dict["columns"],
+        row_limit=row_limit,
+    )
+
+    stars = vizier.query_constraints(Vmag=f"<{magnitude_cutoff}")[0]
+
+    stars.meta["Catalog"] = StarCatalogues[catalog].value
+
+    return stars
+
+
+def get_bright_stars(
+    time, catalog="Yale", pointing=None, radius=None, magnitude_cut=None
+):
+    """
+    Get an astropy table of bright stars from the Yale bright star catalog
+
+    Parameters
+    ----------
+    time: astropy Time
+        time to which proper motion is applied
+    catalog : string
+        name of the catalog to be used available catalogues are 'Yale' and 'Hipparcos'. Default: 'Yale'
+    pointing: astropy Skycoord
+       pointing direction in the sky (if none is given, full sky is returned)
+    radius: astropy angular units
```

Comment:
`radius: astropy.coordinate.Angle`  ? Is that what you mean by angular units?

Reply:
It is not mandatory Angle (i.e. a Quantity with deg unit is not equiv to Angle)

Reply:
> (i.e. a Quantity with deg unit is not equiv to Angle) 

Angle is a subclass of Quantity that forces angular units and performs additional checks.

Reply:
I'd write `u.Quantity[angle]`